### DISCUSSION TITLE
proper SQLmodel approach for Many to Many subqueryLoad

### DISCUSSION THREAD

**Discussion by SteffRainville** [fastapi/sqlmodel#1508] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get("/user/", response_model=List[UserReadWithClients])
async def user(limit: int = 10, offset: int=0 ):
    with Session(get_engine()) as session:
        statement = select(User).offset(offset).limit(limit).options(subqueryload(User.clients))
        results = session.exec(statement).all()

        return results
```

### Description

I followed the Hero and Team sample it works well but I have a many to many relationship and I get the following message

Parent instance <User at 0x22b6878e880> is not bound to a Session; lazy load operation of attribute 'clients' cannot proceed

I use the sqlAlchem,y orm subquery load option to fix my problem but what is the proper SQLmodel approach for many to many?
(Since it works transparently  for 1 to many relationships)

### Operating System

Windows

### SQLModel Version

0.0.4

### Python Version

3.9.7

**Comment by YuriiMotov** [comment, marked as answer]
I believe the approach with SQLModel is the same as with plain SQLAlchemy here.
So, your solution is valid.

I would only consider using `selectinload` instead of `subqueryload` if there are no specific requirements that prevent you from that
